### ISSUE TITLE
Problem with deploy service using Sub function to to build forwarder arn

### ISSUE THREAD

**Issue by Lukasz-Kalicki** [DataDog/serverless-plugin-datadog#109]
## Expected Behavior
When I run `sls package ` gets an error, lambda function exists on environment, I used before `serverless-pseudo-parameters` to construct forwarder arn  and it worked.
```
datadog:
    forwarder: "arn:aws:lambda:#{AWS::Region}:#{AWS::AccountId}:function:datadog-logs-forwarder"
```
But this package is deprecated, this functionality is now natively supported in Serverless Framework.
So I try to use `Sub` function
```
datadog:
    forwarder: !Sub "arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:datadog-logs-forwarder"
```
## Actual Behavior
When I try to package my service I get an error using Sub(CloudFormation) function to build forwarder arn

## Steps to Reproduce the Problem

  1. Setup serverless framework environment
  1. Deploy forwarder
  1. Try to deploy (separate service) your service using Sub function to build forwarder arn

## Specifications

  - Serverless Framework version: 2.22.0
  - Datadog Serverless Plugin version: 2.15.0
  - Lambda function runtime (Python 3.7, Node 10, etc.):  Node14
  
## Stacktrace
  
  ```
Serverless: Packaging service...
Serverless: [AWS lambda undefined 0.021s 0 retries] getFunction({
  FunctionName: {
    'Fn::Sub': 'arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:datadog-logs-forwarder'
  }
})
 
  Datadog Forwarder Not Found Error ----------------------
 
  DatadogForwarderNotFoundError: Could not perform GetFunction on [object Object].
      at /home/runner/work/my-service/my-service/node_modules/serverless-plugin-datadog/src/forwarder.ts:48:11
      at Generator.throw (<anonymous>)
      at rejected (/home/runner/work/my-service/my-service/node_modules/serverless-plugin-datadog/dist/src/forwarder.js:6:65)
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     Operating System:          linux
     Node Version:              14.15.4
     Framework Version:         2.22.0 (local)
     Plugin Version:            4.4.2
     SDK Version:               2.3.2
     Components Version:        3.6.2
 
Error: Process completed with exit code 1.
  ```

**Comment by Lukasz-Kalicki** [comment]
I tried to build this arn a little differently:
```
  datadog:
    forwarder:
      Fn::Join:
        - ":"
        - - "arn:aws:lambda"
          - Ref: "AWS::Region"
          - Ref: "AWS::AccountId"
          - "function:datadog-logs-forwarder"
```

but the error is very similar:

```
Serverless: Packaging service...
Serverless: [AWS lambda undefined 0.022s 0 retries] getFunction({
  FunctionName: {
    'Fn::Join': [
      ':',
      [
        'arn:aws:lambda',
        { Ref: 'AWS::Region' },
        { Ref: 'AWS::AccountId' },
        'function:datadog-logs-forwarder',
        [length]: 4
      ],
      [length]: 2
    ]
  }
})
 
  Datadog Forwarder Not Found Error ----------------------
 
  DatadogForwarderNotFoundError: Could not perform GetFunction on [object Object].
      at /home/runner/work/my-service/my-service/node_modules/serverless-plugin-datadog/src/forwarder.ts:48:11
      at Generator.throw (<anonymous>)
      at rejected (/home/runner/work/my-service/my-service/node_modules/serverless-plugin-datadog/dist/src/forwarder.js:6:65)
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     Operating System:          linux
     Node Version:              14.15.4
     Framework Version:         2.22.0 (local)
     Plugin Version:            4.4.2
     SDK Version:               2.3.2
     Components Version:        3.6.2
 
Error: Process completed with exit code 1.

```

**Comment by Lukasz-Kalicki** [comment]
So i think the problem is in https://github.com/DataDog/serverless-plugin-datadog/blob/master/src/forwarder.ts in 46 line 
```
await aws.request("Lambda", "getFunction", { FunctionName: functionArn });
```

Here we expect arn as a string, but using the `Sub` function we get [object Object]

**Comment by tianchu** [comment]
@Lukasz-Kalicki I think our current logic does assume that we know the exact ARN of the Forwarder Lambda function, I suspect it's a non-trivial change to support AWS pseudo parameters.

However, I think you can workaround this issue by defining the subscription filter as a CloudFormation resource on your own in the serverless.yml (unset the `datadog` `forwarder` parameter) https://docs.datadoghq.com/logs/guide/send-aws-services-logs-with-the-datadog-lambda-function/?tab=cloudformation#collecting-logs-from-cloudwatch-log-group

**Comment by Bolik777** [comment]
it worked perfectly with the previous versions. Currently i use v2.13.1 and it works, so something was broken in the newest release.

**Comment by DarcyRaynerDD** [comment]
Hi. We're investigating a fix for this. It appears to be a regression introduced with PR #105

**Comment by hjiang27** [comment]
Hi! This issue should be fixed in the [new](https://github.com/DataDog/serverless-plugin-datadog/releases/tag/2.16.1) release. Please try it out and let us know if you're still having issues.

**Comment by exiadbq** [comment]
Sorry, seems 2.16.1 with serverless 1.83.2 is not working
```
Serverless: [AWS lambda undefined 0.008s 0 retries] getFunction({
  FunctionName: { 'Fn::ImportValue': 'datadog-forwarder-ForwarderArn' }
})
 
  Datadog Forwarder Not Found Error ----------------------
 
  DatadogForwarderNotFoundError: Could not perform GetFunction on [object Object].
      at /Users/raydai/Development/gs/events-activator/node_modules/serverless-plugin-datadog/src/forwarder.ts:48:11
      at Generator.throw (<anonymous>)
      at rejected (/Users/raydai/Development/gs/events-activator/node_modules/serverless-plugin-datadog/dist/src/forwarder.js:6:65)
  From previous event:
      at PluginManager.invoke (/Users/raydai/Development/gs/events-activator/node_modules/serverless/lib/classes/PluginManager.js:498:22)
      a
```

**Comment by sfirrin** [comment]
Hi @exiadbq, thanks for reporting this. I was able to reproduce and found that the fix that was meant to be in 2.16.1 was not included in the release published to NPM by mistake.

I've just made a new release [2.16.2](https://github.com/DataDog/serverless-plugin-datadog/releases/tag/v2.16.2) and confirmed that using that version fixes the issue on my machine. Could you try the new version out and let us know if you're still seeing the problem? Instead of this error you should now see a log message that says `Serverless: Skipping forwarder ARN validation because forwarder string defined with CloudFormation function`. 

cc @Lukasz-Kalicki @Bolik777

**Comment by Lukasz-Kalicki** [comment]
Hi, with version 2.16.2 `Sub` function works correctly, good work!
